app/sections/collection-header.tsx

```diff
@@ -0,0 +1,142 @@
+import type {
+    HydrogenComponentProps,
+    HydrogenComponentSchema,
+} from '@weaverse/hydrogen';
+import { forwardRef } from 'react';
+import { useLoaderData } from '@remix-run/react';
+import { CollectionDetailsQuery } from 'storefrontapi.generated';
+
+interface HeaderProps extends HydrogenComponentProps {
+    heightSectionDesktop: number;
+    heightSectionMobile: number;
+    enableBackground: boolean;
+    textPosition: string;
+    overlayBackground: number;
+    enableOverlay: boolean;
+}
+
+let CollectHeader = forwardRef<HTMLElement, HeaderProps>((props, ref) => {
+    let { heightSectionDesktop, heightSectionMobile, enableBackground, textPosition, overlayBackground, enableOverlay, ...rest } = props;
+    let { collection } = useLoaderData<
+        CollectionDetailsQuery & {
+            collections: Array<{ handle: string; title: string }>;
+        }
+    >();
+    let backgroundStyle = {};
+    if (enableBackground) {
+        backgroundStyle = {
+            backgroundImage: `url('${collection?.image?.url}')`,
+            '--header-height-desktop': `${heightSectionDesktop}px`,
+            '--header-height-mobile': `${heightSectionMobile}px`
+        };
+    }
+    let textStyle = {};
```

Comment:
I think using class name instead of inline style to set the text color is better here